### PR TITLE
Upgrade to support relay v10.0.1

### PR DESCRIPTION
**Pull request by thicodes**
- [x] Sync Fixtures
 - [x] Update examples

### CODE REVIEW

**Comment by zephraph** [comment]
Given that this contains some major updates to direct dependencies should we release a major? cc @alloy.

File: test/__snapshots__/TypeScriptGenerator-test.ts.snap [hunk 1]
```diff
@@ -3789,8 +4047,8 @@ export type TestDeferRawResponse = {
             }) | null;
         }) | null;
     } | {
-        readonly __typename: string | null;
-        readonly id: string | null;
+        readonly __typename: string;
+        readonly id: string;
```
Comment (alloy) [inline test/__snapshots__/TypeScriptGenerator-test.ts.snap:4051]: Why did all the nullability of these fields change? Did the test schema change?

Comment (maraisr) [reply test/__snapshots__/TypeScriptGenerator-test.ts.snap:4051]: I think its okay to merge this snapshot, the field is non-nullable in the testschema https://github.com/facebook/relay/blob/master/packages/relay-test-utils-internal/testschema.graphql#L813

Test utils was also updated: https://github.com/relay-tools/relay-compiler-language-typescript/pull/199/files#diff-8ee2343978836a779dc9f8d6b794c3b2R5104

I can;t exactly find a commit to test utils that touched the null-ability of that field. But I think its safe to push it.

Comment (alloy) [reply test/__snapshots__/TypeScriptGenerator-test.ts.snap:4051]: Yeah if the schema has these fields as non-nullable, then that’s good enough for me 👍

**Comment by alloy** [comment]
:rocket: PR was released in v13.0.0 :rocket:

**Pull request merged**